### PR TITLE
Remove deprecated JVM option from Dockerfile 

### LINKED ISSUE #2596: JVM opts -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap are deprecated
Since about 8u191, the experimentalVMOption alternatives have been backported from 10, and `UseContainerSupport` is on by default. The options currently set are deprecated. Users wishing to fine tune the memory usage should use the **float** options `InitialRAMPercentage` and `MaxRAMPercentage`, or set mx/ms explicitly.
E.g. `docker run -m 1GB openjdk:8-slim java  -XX:MaxRAMPercentage=80.0`. Failing to specify the trailing `.0` is an error in jdk<10.

`ENV JAVA_OPTS -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap` @
https://github.com/zeebe-io/zeebe/blob/develop/Dockerfile#L10 should be removed.

https://bugs.openjdk.java.net/browse/JDK-8196595

```
# setting mx to .8GB
# sudo docker run -m 1GB openjdk:8-slim java  -XX:+PrintFlagsFinal -mx819m 2>/dev/null | grep -iP "UseContainerSupport | *RAMPercentage | *HeapSize" | sort
     bool UseContainerSupport                       = true                                {product}
   double InitialRAMPercentage                      = 1.562500                            {product}
   double MaxRAMPercentage                          = 25.000000                           {product}
   double MinRAMPercentage                          = 50.000000                           {product}
    uintx ErgoHeapSizeLimit                         = 0                                   {product}
    uintx HeapSizePerGCThread                       = 87241520                            {product}
    uintx InitialHeapSize                          := 16777216                            {product}
    uintx LargePageHeapSizeThreshold                = 134217728                           {product}
    uintx MaxHeapSize                              := 859832320                           {product}
# setting 80%
# sudo docker run -m 1GB openjdk:8-slim java  -XX:+PrintFlagsFinal -XX:MaxRAMPercentage=80.0 2>/dev/null | grep -iP "UseContainerSupport | *RAMPercentage | *HeapSize" | sort
     bool UseContainerSupport                       = true                                {product}
   double InitialRAMPercentage                      = 1.562500                            {product}
   double MaxRAMPercentage                         := 80.000000                           {product}
   double MinRAMPercentage                          = 50.000000                           {product}
    uintx ErgoHeapSizeLimit                         = 0                                   {product}
    uintx HeapSizePerGCThread                       = 87241520                            {product}
    uintx InitialHeapSize                          := 16777216                            {product}
    uintx LargePageHeapSizeThreshold                = 134217728                           {product}
    uintx MaxHeapSize                              := 859832320                           {product}
# setting nothing defaults to 25% RAM
# sudo docker run -m 1GB openjdk:8-slim java  -XX:+PrintFlagsFinal 2>/dev/null | grep -iP "UseContainerSupport | *RAMPercentage | *HeapSize" | sort
     bool UseContainerSupport                       = true                                {product}
   double InitialRAMPercentage                      = 1.562500                            {product}
   double MaxRAMPercentage                          = 25.000000                           {product}
   double MinRAMPercentage                          = 50.000000                           {product}
    uintx ErgoHeapSizeLimit                         = 0                                   {product}
    uintx HeapSizePerGCThread                       = 87241520                            {product}
    uintx InitialHeapSize                          := 16777216                            {product}
    uintx LargePageHeapSizeThreshold                = 134217728                           {product}
    uintx MaxHeapSize                              := 268435456                           {product}

```

Notes:
* Setting Xmx overrides this setting, and can be used in kube land along with `requests`. The actual memory usage (heap + stack + whatever) can be more than Xmx.
* the jdk will almost never use MinRAMPercentage - only if memory is <96MB
```
  // If the maximum heap size has not been set with -Xmx,
  // then set it as fraction of the size of physical memory,
  // respecting the maximum and minimum sizes of the heap.
  if (FLAG_IS_DEFAULT(MaxHeapSize)) {
    julong reasonable_max = (julong)((phys_mem * MaxRAMPercentage) / 100);
    const julong reasonable_min = (julong)((phys_mem * MinRAMPercentage) / 100);
    if (reasonable_min < MaxHeapSize) {
      // Small physical memory, so use a minimum fraction of it for the heap
      reasonable_max = reasonable_min;
    } else {
      // Not-small physical memory, so require a heap at least
      // as large as MaxHeapSize
      reasonable_max = MAX2(reasonable_max, (julong)MaxHeapSize);
    }
```

### PR DESCRIPTION
**Pull request by afirth**
and replace with MaxRAMPercentage

closes #2596

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 762c71cf1df081adc14de2b8affabae024cb5529
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

File: Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ ENV ZB_HOME=/usr/local/zeebe \
     ZEEBE_GATEWAY_HOST=0.0.0.0 \
     ZEEBE_STANDALONE_GATEWAY=false
 ENV PATH "${ZB_HOME}/bin:${PATH}"
-ENV JAVA_OPTS -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap
+ENV JAVA_OPTS -XX:MaxRAMPercentage:80.0
```
Comment (menski) [inline Dockerfile]: I guess this should be `MaxRAMPercentage=80.0`, i.e. `=` instead of `:`?

**Review by menski — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (afirth) [reply Dockerfile]: fixed, with my sincere apologies

**Review by menski — APPROVED** [review]

**Comment by menski** [comment]
Thanks :+1: 

bors r+

**Comment by menski** [comment]
I rebuild and pushed the 0.18.0/latest tag for the docker hub image

**Pull request merged**